PWGHF/TableProducer/candidateSelectorDsToKKPi.cxx

```diff
@@ -309,11 +315,25 @@ struct HfCandidateSelectorDsToKKPi {
       }
 
       // track-level PID selection
-      int pidTrackPos1Pion = selectorPion.statusTpcOrTof(trackPos1);
-      int pidTrackPos1Kaon = selectorKaon.statusTpcOrTof(trackPos1);
-      int pidTrackPos2Pion = selectorPion.statusTpcOrTof(trackPos2);
-      int pidTrackPos2Kaon = selectorKaon.statusTpcOrTof(trackPos2);
-      int pidTrackNegKaon = selectorKaon.statusTpcOrTof(trackNeg);
+      int pidTrackPos1Pion = -1;
+      int pidTrackPos1Kaon = -1;
+      int pidTrackPos2Pion = -1;
+      int pidTrackPos2Kaon = -1;
+      int pidTrackNegKaon = -1;
+
+      if (usePidTpcAndTof) {
+        selectorPion.statusTpcAndTof(trackPos1);
+        selectorKaon.statusTpcAndTof(trackPos1);
+        selectorPion.statusTpcAndTof(trackPos2);
+        selectorKaon.statusTpcAndTof(trackPos2);
+        selectorKaon.statusTpcAndTof(trackNeg);
+      } else {
+        selectorPion.statusTpcOrTof(trackPos1);
+        selectorKaon.statusTpcOrTof(trackPos1);
+        selectorPion.statusTpcOrTof(trackPos2);
+        selectorKaon.statusTpcOrTof(trackPos2);
+        selectorKaon.statusTpcOrTof(trackNeg);
+      }
```

Comment:
You are not assigning the results to the variables.

Reply:
Ops, you're right sorry!